core/src/main/java/org/apache/iceberg/rest/RESTSerializers.java

```diff
@@ -75,7 +78,12 @@ public static void registerAll(ObjectMapper mapper) {
         .addSerializer(UpdateRequirement.class, new UpdateRequirementSerializer())
         .addDeserializer(UpdateRequirement.class, new UpdateRequirementDeserializer())
         .addSerializer(OAuthTokenResponse.class, new OAuthTokenResponseSerializer())
-        .addDeserializer(OAuthTokenResponse.class, new OAuthTokenResponseDeserializer());
+        .addDeserializer(OAuthTokenResponse.class, new OAuthTokenResponseDeserializer())
+        .addSerializer(ReportMetricsRequest.class, new ReportMetricsRequestSerializer<>())
+        .addDeserializer(ReportMetricsRequest.class, new ReportMetricsRequestDeserializer<>())
+        .addSerializer(ImmutableReportMetricsRequest.class, new ReportMetricsRequestSerializer<>())
```

Comment:
I'll look into this in a follow-up. I was expecting that registering a custom serializer/deserializer for `ReportMetricsRequest` would also work out-of-the-box for `ImmutableReportMetricsRequest` but apparently that's not the case and it requires the subclass to be registered explicitly

Reply:
I created https://github.com/apache/iceberg/pull/5911 to address this